### PR TITLE
Add unsafeBitCast documentation to the Standard Library Programmer's …

### PR DESCRIPTION
**Pull request by atrick**
…Guide

Explaining the many other ways that one can cast references.

### CODE REVIEW

**Comment by atrick** [comment]
@gottesmm I put you on the reviewer list since you requested this!

**Review by gottesmm — APPROVED** [review]
Looks great!

File: docs/StandardLibraryProgrammersManual.md [hunk 1]
```diff
@@ -50,6 +50,19 @@ TODO: Should this subsume or link to [AccessControlInStdlib.rst](https://github.
...
+
+Reference casting is more interesting. References casting can include converting to an Optional reference and converting from a class constrained existential.
+
+The regular `as` operator should be able to convert between reference types with full dynamic checking.
+
+`unsafeDownCast` is just as capable, but is only dynamically checked in debug mode or if the cast requires runtime support.
+
+`_unsafeUncheckedDowncast` is the same but is only dynamically checked in the stdlib asserts build, or if the cast requires runtime support.
+
+`_unsafeReferenceCast` is only dynamically checked if the cast requires runtime support. Additionally, it does not impose any static `AnyObject` constraint on the incoming reference. This is useful in a generic context where the object-ness can be determined dynamically, as done is some bridged containers.
```
Comment (milseman) [inline docs/StandardLibraryProgrammersManual.md]: >  as done is some bridged containers.

typo in there.

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Pull request merged**